constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/widget/MotionLayout.java

```diff
@@ -1888,15 +1890,24 @@ public void touchAnimateTo(int touchUpMode, float position, float currentVelocit
 
         switch (touchUpMode) {
             case TOUCH_UP_COMPLETE:
+            case TOUCH_UP_NEVER_TO_START:
+            case TOUCH_UP_NEVER_TO_END:
             case TOUCH_UP_COMPLETE_TO_START:
             case TOUCH_UP_COMPLETE_TO_END: {
-                if (touchUpMode == TOUCH_UP_COMPLETE_TO_START) {
+                if (touchUpMode == TOUCH_UP_COMPLETE_TO_START || touchUpMode == TOUCH_UP_NEVER_TO_END) {
                     position = 0;
-                } else if (touchUpMode == TOUCH_UP_COMPLETE_TO_END) {
+                } else if (touchUpMode == TOUCH_UP_COMPLETE_TO_END || touchUpMode == TOUCH_UP_NEVER_TO_START) {
                     position = 1;
                 }
-                mStopLogic.config(mTransitionLastPosition, position, currentVelocity,
-                        mTransitionDuration, mScene.getMaxAcceleration(), mScene.getMaxVelocity());
+                float stiff = mScene.getSpringStiffiness();
+                if(Float.isNaN(stiff)) {
```

Comment:
hmm should you instead check if the engine is the spring engine?

Reply:
This is configuring the stop logic as a spring.
It decides this by checking if stiffness is a number.